tempodb/backend/s3/s3.go

```diff
@@ -744,6 +748,14 @@ func buildSSEConfig(cfg *Config) (encrypt.ServerSide, error) {
 		}
 	case SSES3:
 		return encrypt.NewSSE(), nil
+	case SSEC:
+		if cfg.SSE.CustomerEncryptionKey == "" {
+			return nil, errors.New("SSE-C EncryptionKey is missing")
+		}
+		if len(cfg.SSE.CustomerEncryptionKey) != 32 {
+			return nil, errors.New("SSE-C EncryptionKey must be 32 bytes long")
```

Comment:
Minio already validates the key. Maybe we can rely on that

https://github.com/minio/minio-go/blob/785b638cc4afc979cb92a1358f172d05d32a18bf/pkg/encrypt/server-side.go#L115